internal/services/setup_wallets_for_network_service.go

```diff
@@ -117,10 +96,68 @@ func SetupWalletsForProperNetwork(ctx context.Context, dbConnectionPool db.DBCon
 			return fmt.Errorf("error upserting wallets: %w", err)
 		}
 
+		// 2. Upsert Wallet Assets (if any)
+		models, err := data.NewModels(dbConnectionPool)
+		if err != nil {
+			return fmt.Errorf("getting models: %w", err)
+		}
+
+		// 2.1. Fetch all assets and populate a map for quick lookup
+		assets, err := models.Assets.GetAll(ctx)
+		if err != nil {
+			return fmt.Errorf("getting all available assets on database: %w", err)
+		}
+		existingAssets := make(map[string]string)
+		for _, asset := range assets {
+			existingAssets[asset.Code+":"+asset.Issuer] = asset.ID
+		}
+
+		// 2.2. Populate assets
+		walletNames := []string{}
+		assetCodes := []string{}
+		assetIssuers := []string{}
+
+		for _, wallet := range wallets {
+			for _, asset := range wallet.Assets {
+				if _, exists := existingAssets[asset.Code+":"+asset.Issuer]; exists {
+					walletNames = append(walletNames, wallet.Name)
+					assetCodes = append(assetCodes, asset.Code)
+					assetIssuers = append(assetIssuers, asset.Issuer)
+				}
+			}
+		}
+
+		if len(walletNames) == 0 {
+			log.Ctx(ctx).Info("no assets to be inserted for the given wallets")
+			return nil
+		}
+
+		query = `
+			WITH wallets_assets_to_insert AS (
+				SELECT 
+					UNNEST($1::TEXT[]) as wallet_name, 
+					UNNEST($2::TEXT[]) as asset_code, 
+					UNNEST($3::TEXT[]) as asset_issuer
+			)
+			, wallet_asset_ids AS (
+				SELECT w.id as wallet_id, a.id as asset_id
+				FROM wallets_assets_to_insert wa
+				JOIN wallets w ON wa.wallet_name = w.name
+				JOIN assets a ON wa.asset_code = a.code AND wa.asset_issuer = a.issuer
+			)
+			INSERT INTO wallets_assets(wallet_id, asset_id)
+			SELECT wallet_id, asset_id FROM wallet_asset_ids
+			ON CONFLICT(wallet_id, asset_id) DO NOTHING;
+		`
+
+		_, err = dbTx.ExecContext(ctx, query, pq.Array(walletNames), pq.Array(assetCodes), pq.Array(assetIssuers))
+		if err != nil {
+			return fmt.Errorf("upserting wallet assets: %w", err)
+		}
```

Comment:
💡 WDYT about logging the number of rows affected from this query execution?

Reply:
Addressed in 418d55a8854974dd676e3f6140980747d9e79a56